### PR TITLE
adds hardware signer to local server ec middleware, moar tests

### PR DESCRIPTION
**Pull request by James-Pickett**

### CODE REVIEW

**Review by directionless — COMMENTED** [review]
Seems fine, though I think it'll have some conflicts with #1020

File: ee/localserver/krypto-ec-middleware.go [hunk 1]
```diff
@@ -27,16 +27,17 @@ type v2CmdRequestType struct {
...
 
-func newKryptoEcMiddleware(logger log.Logger, signer crypto.Signer, counteParty ecdsa.PublicKey) *kryptoEcMiddleware {
+func newKryptoEcMiddleware(logger log.Logger, localDbSigner, hardwareSigner crypto.Signer, counteParty ecdsa.PublicKey) *kryptoEcMiddleware {
 	return &kryptoEcMiddleware{
-		signer:       signer,
-		counterParty: counteParty,
-		logger:       log.With(logger, "keytype", "ec"),
+		localDbSigner:  localDbSigner,
+		hardwareSigner: hardwareSigner,
+		counterParty:   counteParty,
```
Comment (seejdev) [inline ee/localserver/krypto-ec-middleware.go:39]: Not your code, but `counteParty` typo

Comment (James-Pickett) [reply ee/localserver/krypto-ec-middleware.go]: it is my code, just part of this pr =), fixed

**Review by directionless — APPROVED** [review]
Let's try

**Pull request merged**